### PR TITLE
Improve NEST GPU Utilization 2/N

### PR DESCRIPTION
**Pull request by MahmoudAshraf97**
This PR is among a series aimed to improve the training speed and GPU utilization of NEST models #13619 

after profiling, the masking module takes a good chunk of the total training loop time while involving no computation, thus reducing the utilization, this PR reduces python for-loops as much as possible, and converts the masking operation to a singe function call

As shown in the following graph, This PR along with #14086 reduces the training step time by almost 17%
{4751020E-438F-4DA7-AE1F-E6F14653A76A}

The green is with the PRs applied

cc @stevehuang52

### CODE REVIEW

File: nemo/collections/asr/modules/ssl_modules/masking.py [hunk 1]
```diff
@@ -99,29 +98,42 @@ def forward_without_overlap(self, input_feats: torch.Tensor, input_lengths: torc
             masks (Tensor): the generated masks, shape=(batch, features, time)
         """
         batch_size = input_feats.size(0)
-        mask_value = self.mask_embedding.unsqueeze(-1)
         masks = torch.zeros_like(input_feats)
-        maksed_feats = input_feats.clone()
+        masked_feats = input_feats
```
Comment (stevehuang52) [inline nemo/collections/asr/modules/ssl_modules/masking.py:102]: Will this also change `input_feats` as well when `masked_feats` is updated? Ideally we should return the `masked_feats` while keeping `input_feats` unchanged.

Comment (MahmoudAshraf97) [reply nemo/collections/asr/modules/ssl_modules/masking.py:102]: I did confirm that by:
```python
module = RandomBlockMasking(80, allow_overlap=False, mask_prob=0.01, block_size=40).cuda()
for i in range(100):
    input = torch.randn(1, 80, 1000).cuda()
    input_len = torch.tensor([1000]).cuda()
    masked_feats, masks = module(input, input_len)
    assert not torch.allclose(masked_feats, input)
```

Comment (stevehuang52) [reply nemo/collections/asr/modules/ssl_modules/masking.py:102]: Awesome, thanks

**Comment by stevehuang52** [comment]
@MahmoudAshraf97 could you please fix the DCO error?

**Comment by MahmoudAshraf97** [comment]
@stevehuang52 Fixed

**Comment by MahmoudAshraf97** [comment]
@stevehuang52 Can you rerun CICD?

**Review by stevehuang52 — APPROVED** [review]
Looks good to me, thanks for the improvement!
